Source/AwsCommonRuntimeKit/crt/AWSString.swift

```diff
@@ -3,38 +3,31 @@
 import AwsCCommon
 
 final class AWSString {
-    let string: String
     let rawValue: UnsafeMutablePointer<aws_string>
 
     init(_ str: String, allocator: Allocator) {
-        self.string = str
         self.rawValue = aws_string_new_from_array(allocator.rawValue, str, str.count)
     }
 
     var count: Int {
         return self.rawValue.pointee.len
     }
 
-    func newByteCursor() -> ByteCursor {
-        return AWSStringByteCursor(self)
-    }
-
-    func asCStr() -> UnsafePointer<Int8> {
-        return aws_string_c_str(self.rawValue)
-    }
-
     deinit {
         aws_string_destroy(self.rawValue)
     }
 }
 
-private struct AWSStringByteCursor: ByteCursor {
-    private let awsString: AWSString
-    public var rawValue: aws_byte_cursor
+// TODO: try to find a better way to get a long lived byte cursor from String without making a copy.
+class AWSStringByteCursor {
+    let awsString: AWSString
+    let string: String
```

Comment:
Storing String doesn’t occur any cost because Swift doesn’t make a copy until it is mutated. I needed it in the HttpProxyOption class because of complexity of didSet. I will get rid of it after the revamp of withCStruct or getCStruct.

Reply:
Thanks, fixed.